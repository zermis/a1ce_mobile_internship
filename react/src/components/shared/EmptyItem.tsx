import React from 'react';
import { Image, StyleSheet, Text, View } from 'react-native';

interface Props {
    header?: string;
}

const EmptyItem = (props: Props): JSX.Element => {
    return (
        <View style={styles.container}>
            {props.header && (
                <Text style={styles.headerText}>{props.header}</Text>
            )}
            <Image
                style={styles.image}
                source={require('../../assets/images/panda.png')}
            ></Image>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        paddingTop: 20,
        overflow: 'hidden',
        alignItems: 'center'
    },
    headerText: {
        fontSize: 18,
        fontFamily: 'ibm-regular'
    },
    image: {
        height: 220,
        width: 250,
        alignSelf: 'center'
    }
});

export default EmptyItem;
