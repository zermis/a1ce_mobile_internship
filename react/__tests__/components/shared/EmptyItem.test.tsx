import EmptyItem from '../../../src/components/shared/EmptyItem';
import renderer from 'react-test-renderer';

it('render empty item', () => {
    const tree = renderer.create(<EmptyItem header="No items" />).toJSON();
    expect(tree).toMatchSnapshot();
});
