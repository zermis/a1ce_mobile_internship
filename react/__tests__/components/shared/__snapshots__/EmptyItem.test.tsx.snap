// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render empty item 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "overflow": "hidden",
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "ibm-regular",
        "fontSize": 18,
      }
    }
  >
    No items
  </Text>
  <Image
    source={1}
    style={
      Object {
        "alignSelf": "center",
        "height": 220,
        "width": 250,
      }
    }
  />
</View>
`;
